File: app/src/androidTest/java/com/twitli/android/twitter/dagger/TestModule.java
package com.twitli.android.twitter.dagger;

import dagger.Module;
import dagger.Provides;
import org.mockito.Mockito;

@Module
public class TestModule {

    private TwitManager twitManager;

    @Provides
    public WikiPageManager provideWikiPageManager() {
        return Mockito.mock(WikiPageManager.class);
    }

    @Provides
    public TwitManager provideTwitManager() {
        if(twitManager == null){
            twitManager = Mockito.mock(TwitManager.class);
        }
        return twitManager;
    }

}

File: app/src/androidTest/java/com/twitli/android/twitter/dagger/TestModule.kt
package com.twitli.android.twitter.dagger

import com.twitli.android.twitter.bot.ChatBot
import com.twitli.android.twitter.bot.wiki.WiktionaryBot
import com.twitli.android.twitter.bot.wiki.api.MyHttp
import com.twitli.android.twitter.bot.wiki.api.MyHttpImpl
import com.twitli.android.twitter.bot.wiki.api.WiktionaryApi
import com.twitli.android.twitter.bot.wiki.api.WiktionaryApiImpl
import com.twitli.android.twitter.tweet.TwitManager
import com.twitli.android.twitter.wiki.WikiPageManager
import dagger.Module
import dagger.Provides
import org.mockito.Mockito

@Module
open class TestModule {

    lateinit var twitManager: TwitManager
    lateinit var wikiPageManager: WikiPageManager

    @Provides
    fun provideTwitManager(): TwitManager {
        if (!this::twitManager.isInitialized) run {
            twitManager = Mockito.mock(TwitManager::class.java)
        }
        return twitManager
    }
}
